git-tar/function/ops.go

```diff
@@ -43,18 +44,28 @@ func parseYAML(pushEvent sdk.PushEvent, filePath string) (*stack.Services, error
 }
 
 func fetchTemplates(filePath string) error {
-	templateRepos := []string{"https://github.com/openfaas/templates", "https://github.com/openfaas-incubator/node8-express-template.git", "https://github.com/openfaas-incubator/golang-http-template.git"}
-
-	for _, repo := range templateRepos {
-		pullCmd := exec.Command("faas-cli", "template", "pull", repo)
-		pullCmd.Dir = filePath
-		err := pullCmd.Start()
-		if err != nil {
-			return fmt.Errorf("Failed to start faas-cli template pull: %t", err)
+	if _, err := os.Stat(filePath + "/template"); os.IsNotExist(err) {
+		var eg errgroup.Group
+		templateRepos := []string{"https://github.com/openfaas/templates", "https://github.com/openfaas-incubator/node8-express-template.git", "https://github.com/openfaas-incubator/golang-http-template.git"}
+
+		for _, repo := range templateRepos {
```

Comment:
@AkihiroSuda please could you review this use of errgroup?

Reply:
Almost looks good but why ignore non-nil err from eg.Wait?

Reply:
Ah ... Thanks for the feedback @AkihiroSuda I forgot I missed the bad error case ..